exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/DrillFunctionRegistry.java

```diff
@@ -92,38 +94,110 @@ public DrillFunctionRegistry(ScanResult classpathScan) {
-            op = new DrillSqlAggOperator(name, func.getParamCount());
-          } else {
-            boolean isDeterministic;
-            // prevent Drill from folding constant functions with types that cannot be materialized
-            // into literals
-            if (DrillConstExecutor.NON_REDUCIBLE_TYPES.contains(func.getReturnType().getMinorType())) {
-              isDeterministic = false;
-            } else {
-              isDeterministic = func.isDeterministic();
-            }
-            op = new DrillSqlOperator(name, func.getParamCount(), func.getReturnType(), isDeterministic);
-          }
-          operatorTable.add(function.getKey(), op);
+        final int paramCount = func.getParamCount();
+        if(func.isAggregating()) {
+          aggregateFunctions.put(paramCount, func);
+        } else {
+          final Pair<Integer, Integer> argNumerRange = getArgNumerRange(name, func);
+          functions.put(argNumerRange, func);
         }
+
+        if(!func.isDeterministic()) {
+          isDeterministic = false;
+        }
+      }
+      for (Entry<Pair<Integer, Integer>, Collection<DrillFuncHolder>> entry : functions.asMap().entrySet()) {
+        final DrillSqlOperator drillSqlOperator;
+        final Pair<Integer, Integer> range = entry.getKey();
+        final int max = range.getRight();
+        final int min = range.getLeft();
+        drillSqlOperator = new DrillSqlOperator(
+            name,
+            Lists.newArrayList(entry.getValue()),
+            min,
+            max,
+            isDeterministic);
+        operatorTable.add(name, drillSqlOperator);
+      }
+      for (Entry<Integer, Collection<DrillFuncHolder>> entry : aggregateFunctions.asMap().entrySet()) {
+        operatorTable.add(name, new DrillSqlAggOperator(name, Lists.newArrayList(entry.getValue()), entry.getKey()));
       }
     }
+
+    registerCalcitePlaceHolderFunction(operatorTable);
+  }
+
+  /**
+   * These {@link DrillSqlOperator} merely act as a placeholder so that Calcite
+   * allows convert_to(), convert_from(), flatten(), date_part() functions in SQL.
+   */
+  private void registerCalcitePlaceHolderFunction(DrillOperatorTable operatorTable) {
+    final String convert_to = "CONVERT_TO";
+    final String convert_from = "CONVERT_FROM";
+    final String flatten = "FLATTEN";
+    final String date_part = "DATE_PART";
+
+    operatorTable.add(convert_to,
+        new DrillSqlOperator(convert_to,
+            2,
+            true));
+    operatorTable.add(convert_from,
+        new DrillSqlOperator(convert_from,
+            2,
+            true));
+    operatorTable.add(flatten,
+        new DrillSqlOperator(flatten,
+            1,
+            true));
+    operatorTable.add(date_part,
+        new DrillSqlOperator(date_part,
+            2,
+            true));
   }
 
+  private Pair<Integer, Integer> getArgNumerRange(final String name, final DrillFuncHolder func) {
+    switch(name.toUpperCase()) {
+      case "CONCAT":
```

Comment:
@jacques-n Yes, once per registry. 

This patch has not changed that behavior.